### PR TITLE
Cleanup variable usage, use testing matrix, reduce build times.

### PR DESCRIPTION
**Pull request by fredrikhgrelland**

### CODE REVIEW

File: ansible/bootstrap.yml [hunk 1]
```diff
@@ -86,31 +90,13 @@
 
     - name: Run the nomad acl bootstrap process
       include_tasks: nomad_acl_bootstrap.yml
-      when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"
+      when: lookup('env', 'nomad_acl') | bool
 
     - name: Create roles for nomad in vault
       include_tasks: vault_nomad_integration.yml
-      when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"
```
Comment (pdmthorsrud) [inline ansible/bootstrap.yml:93]: The `when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"` should not be removed. It makes no sense to run a vault - nomad integration without ACLs enabled in nomad. WIthout ACLs enabled there are no policies in nomad to create roles for etc..

File: ansible/bootstrap.yml [hunk 2]
```diff
@@ -86,31 +90,13 @@
 
     - name: Run the nomad acl bootstrap process
       include_tasks: nomad_acl_bootstrap.yml
-      when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"
+      when: lookup('env', 'nomad_acl') | bool
 
     - name: Create roles for nomad in vault
       include_tasks: vault_nomad_integration.yml
```
Comment (pdmthorsrud) [inline ansible/bootstrap.yml:96]:
```suggestion
      include_tasks: vault_nomad_integration.yml
      when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"
```
It makes no sense to run a vault - nomad integration without ACLs enabled in nomad. WIthout ACLs enabled there are no policies in nomad to create roles for etc..

File: ansible/templates/consul-acl.hcl.j2 [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {% raw -%}
 acl {
-  default_policy = "deny"
   down_policy = "extend-cache"
   enabled = true
   tokens {
     master = "master"
   }
+  {% endraw %}
+  default_policy = "{{ lookup('env', 'consul_acl_default_policy') }}"
```
Comment (pdmthorsrud) [inline ansible/templates/consul-acl.hcl.j2:9]: Just out of curiosity, is this after the raw section because it is jinja2-templating?

Comment (fredrikhgrelland) [reply ansible/templates/consul-acl.hcl.j2:9]: Yep

File: ansible/bootstrap.yml [hunk 4]
```diff
@@ -86,31 +90,14 @@
...
-      when: "{{ lookup('env', 'TF_VAR_nomad_acl') }}"
+      when: lookup('env', 'nomad_acl') | bool
 
     - name: Vault enable consul secret engine
       include_tasks: vault_consul_integration.yml
-      when: "'{{ lookup('env', 'consul_acl_default_policy') }}' == 'deny'"
```
Comment (Neha-Sinha2305) [inline ansible/bootstrap.yml:97]: Shouldn't we include this check here:
when: lookup('env', 'consul_acl') | bool

Comment (fredrikhgrelland) [reply ansible/bootstrap.yml:97]: Yes. I do not know if this breaks anything, but will push an update.

**Comment by fredrikhgrelland** [comment]
@pdmthorsrud What about this? https://github.com/fredrikhgrelland/vagrant-hashistack/runs/966481996?check_suite_focus=true

File: ansible/bootstrap.yml (hunk 2 above)
Comment (zhenik) [reply ansible/bootstrap.yml]: The same statement applies on `consul secrets engine`

**Pull request merged**